src/analysis/calculus/mean_value.lean

```diff
@@ -896,12 +896,32 @@ begin
 end
 
 /-- If a function `f` is continuous on a convex set `D ⊆ ℝ`, is differentiable on its interior,
-and `f'` is monotone on the interior, then `f` is convex on `ℝ`. -/
+and `f'` is antimonotone on the interior, then `f` is concave on `D`. -/
+theorem concave_on_of_deriv_antimono {D : set ℝ} (hD : convex D) {f : ℝ → ℝ}
+  (hf : continuous_on f D) (hf' : differentiable_on ℝ f (interior D))
+  (hf'_mono : ∀ x y ∈ interior D, x ≤ y → deriv f y ≤ deriv f x) :
+  concave_on D f :=
```

Comment:
How about adding variants of this and `concave_on_of_deriv2_nonpos` and their convex versions, in which `D` is assumed to be open (and then the `interior D` collapses to `D` and `hf` can be dropped)?  Then, later, in `concave_on_log_Iio`, there is no need for the repeated `rw [interior_Iio]`s.

Reply:
Good idea -- done!